python-package/lightgbm/sklearn.py

```diff
@@ -375,9 +375,9 @@ def _construct_dataset(X, y, sample_weight, init_score, group, params):
                 if valid_data[0] is X and valid_data[1] is y:
                     valid_set = train_set
                 else:
-                    valid_weight = None if eval_sample_weight is None else eval_sample_weight.get(i, None)
-                    valid_init_score = None if eval_init_score is None else eval_init_score.get(i, None)
-                    valid_group = None if eval_group is None else eval_group.get(i, None)
+                    valid_weight = None if eval_sample_weight is None else eval_sample_weight[i]
+                    valid_init_score = None if eval_init_score is None else eval_init_score[i]
+                    valid_group = None if eval_group is None else eval_group[i]
```

Comment:
thx, can you also help check corresponding part in Python_API.md to see if need to be changed?